[glenda-1506] iP

glenda-1506 opened the pull request.

glenda-1506 committed:
Rename chatbot

glenda-1506 committed:
Add greet function and exit function

glenda-1506 committed:
Add echo function

glenda-1506 committed:
Add add function and list function

glenda-1506 committed:
Add mark as done function

glenda-1506 committed:
Tweak the code to comply with the basic rules of SE-EDU Java coding standarard

glenda-1506 committed:
Add tasktypes array to categorise the items added

glenda-1506 committed:
Add todo category

glenda-1506 committed:
Add deadline category

glenda-1506 committed:
Add event category

PrinceCatt reviewed (commented):
1. Nice naming of commitments
2. good use of OOP
3. Good naming of variables: all camalCase and go well with requirements
4. Consistent identations

Toby-Yu commented on src/main/java/lovespiritual.java:
```diff
@@ -65,12 +65,25 @@ public static void main(String[] args) {
+                String fullTaskDetails = input.substring(9).trim();
+                String[] taskDetails = fullTaskDetails.split("/by");
+                String taskDescription = taskDetails[0].trim();
+                String by = taskDetails.length > 1 ? taskDetails[1].trim() : "";
```
Using .trim() is a smart solution.

Toby-Yu commented on src/main/java/lovespiritual.java:
```diff
@@ -61,6 +61,16 @@ public static void main(String[] args) {
+                System.out.println(line);
+                System.out.println("Got it. I've added this task:");
+                System.out.println(" [T][ ] " + taskDescription);
+                System.out.println("Now you have " + taskCount + " tasks in the list.");
```
Clear and direct code

Toby-Yu commented on src/main/java/lovespiritual.java:
```diff
@@ -2,17 +2,19 @@
-        int count = 0;
+        String[] tasks = new String[100]; // array of tasks
+        boolean[] isMarked = new boolean[100]; // check if task is marked
+        int taskCount = 0; // count the number of tasks added in the array
```
the variable names looks better now

Toby-Yu commented on src/main/java/lovespiritual.java:
```diff
@@ -22,10 +23,40 @@ public static void main(String[] args) {
+                    System.out.println(line);
+                }
+            } else if (input.startsWith("unmark ")) {
+                String numberString = input.substring(7).trim();
```
The nested loop involves more than 1 level which may slightly violate the coding standard but it looks good

Toby-Yu reviewed (commented):
The documentation is gd and the code is clear

AjayShanker-geek commented on src/main/java/lovespiritual.java:
```diff
@@ -0,0 +1,111 @@
+        String line = "__________________________________________________"; // horizontal line
+        Scanner in = new Scanner(System.in);
+        String[] tasks = new String[100]; // array of tasks
+        boolean[] isMarked = new boolean[100]; // check if task is marked
```
Good Boolean Variable used!

AjayShanker-geek commented on src/main/java/lovespiritual.java:
```diff
@@ -0,0 +1,111 @@
+        while (true) {
+            String input = in.nextLine().trim();
+
+            if (input.equalsIgnoreCase("bye")) {
```
Correct use of K&R style brackets.

AjayShanker-geek replied:
This is hard for the reviewer to review due the nested if else statement (Arrow Head). Adding a switch case and separating each conditions as Methods.

AjayShanker-geek commented on src/main/java/lovespiritual.java:
```diff
@@ -0,0 +1,111 @@
+    public static void main(String[] args) {
+        String line = "__________________________________________________"; // horizontal line
+        Scanner in = new Scanner(System.in);
+        String[] tasks = new String[100]; // array of tasks
```
Good variable name that is easy to understand.

AjayShanker-geek commented on src/main/java/lovespiritual.java:
```diff
@@ -0,0 +1,111 @@
+            } else if (input.equalsIgnoreCase("list")) {
+                System.out.println(line);
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < taskCount; i++) {
```
Great that you use iterator variables only in the for loop.

AjayShanker-geek reviewed (commented):
Overall good, but room to improve on the encapsulation and abstraction.

glenda-1506 committed:
Extract functions and constants to acheive Single Level of Abstraction Per function

glenda-1506 committed:
Add and modify code to handle different input formats

glenda-1506 committed:
Create base Task class

glenda-1506 committed:
Modify Main to use Task class

glenda-1506 committed:
Create Todo class

glenda-1506 committed:
Modify Main to use Todo class

glenda-1506 committed:
Create Deadline class

glenda-1506 committed:
Modify Main to use Deadline class

glenda-1506 committed:
Create Event class

glenda-1506 committed:
Modify Main to use Event class

glenda-1506 committed:
Add lovespiritualException class

glenda-1506 committed:
Add try-catch handling in main function and update methods to throw LovespiritualException

glenda-1506 committed:
Replace addTask function with exception handling for invalid commands

glenda-1506 committed:
Add exception handling to event function for missing inputs

glenda-1506 committed:
Add exception handling to deadline function for missing inputs

glenda-1506 committed:
Add exception handling to todo function for missing inputs

glenda-1506 committed:
Add exception handling to mark and unmark function for invalid inputs

glenda-1506 committed:
Refactore all messages with personalized and unique wording

glenda-1506 committed:
Merge branch 'branch-Level-5'

glenda-1506 committed:
Divide classes into packages

glenda-1506 committed:
Replace arrays with ArrayList for dynamic task handling

glenda-1506 committed:
Replace task addition logic from array to ArrayList

glenda-1506 committed:
Replace task status logic from array to ArrayList

glenda-1506 committed:
Replace task list display logic from array to ArrayList

glenda-1506 committed:
Remove redundant variables and replace arrays to ArrayList

glenda-1506 committed:
Remove redundant addTask method

glenda-1506 committed:
Remove redundant array-based task handling and transition to ArrayList methods

glenda-1506 committed:
Add delete function

glenda-1506 committed:
Replace taskCount with ArrayList in printList method and removed redundant taskCount variable

glenda-1506 committed:
Add message for empty task list

glenda-1506 committed:
Extract delete function

glenda-1506 committed:
Update status of variables from protected to public

glenda-1506 committed:
Replace arrays with ArrayList for dynamic task handling

glenda-1506 committed:
Create saveTasks method

glenda-1506 committed:
Create savedFormat method to save the data in the correct format

glenda-1506 committed:
Create loadTasks method to read the text file when the program starts

glenda-1506 committed:
Create extractTasks method to format the data properly when the program starts

glenda-1506 committed:
Add methods and replace arrays with ArrayList

glenda-1506 committed:
Merge branch 'branch-Level-6'

glenda-1506 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/lovespiritual/lovespiritual.java

glenda-1506 committed:
Create artifect

glenda-1506 committed:
Debug errors which stops the saving of data

thoriritu commented:
Overall, function names and variables have been appropriately named. You could work on the documentation of code through the use of comments, so that it is more readable.

Good job!

thoriritu commented on src/main/java/lovespiritual/lovespiritual.java:
```diff
@@ -0,0 +1,356 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class lovespiritual {
```
To improve abstraction, you can refactor the code by separating task-related logic from the main control flow. One class can handle the application's control logic (user interaction, commands) while another class can be responsible for managing tasks (adding, deleting, marking, saving, etc.).

thoriritu commented on src/main/java/lovespiritual/lovespiritual.java:
```diff
@@ -0,0 +1,356 @@
+                writer.newLine();
+            }
+            writer.close();
+        } catch (Exception e) {
```
You should handle exceptions more specifically to avoid masking actual issues and making debugging harder. Consider catching IOException separately for file operations and only catch Exception for truly unexpected issues.

glenda-1506 committed:
Debug method to load saved file

glenda-1506 committed:
Add saved file

glenda-1506 committed:
Add UI Class

glenda-1506 committed:
Setup environment for print functions

Import Task class from lovespiritual.task
Add seperator constant

glenda-1506 committed:
Refactor printExitScreen() to UI class

glenda-1506 committed:
Refactor printWelcomeScreen() to UI class

glenda-1506 committed:
Refactor printList() to UI class

glenda-1506 committed:
Add printError() function for expected errors

glenda-1506 committed:
Add printUnexpectedError() function for unexpected errors

glenda-1506 committed:
Remove unused constant

glenda-1506 committed:
Refactor functions in main to use those in UI class

glenda-1506 committed:
Add TaskList class

glenda-1506 committed:
Add Storage class

glenda-1506 committed:
Add Parser class

glenda-1506 committed:
Refactor printError method to static and remove redundant line

glenda-1506 committed:
Setup environment for functions in TaskList class

Import classes and library
Add seperator constant

glenda-1506 committed:
Add constructor overload and getter

glenda-1506 committed:
Refactor deleteTask() to taskList class

glenda-1506 committed:
Refactor extractTasks() to taskList class

glenda-1506 committed:
Refactor savedFormat() to TaskList class

glenda-1506 committed:
Import classes and libraries into Storage class

glenda-1506 committed:
Add filePath variable and constructor

glenda-1506 committed:
Refactor saveTasks() to Storage class

glenda-1506 committed:
Refactor loadTasks() to Storage class

glenda-1506 committed:
Remove redundant SEPARATOR print statements

glenda-1506 committed:
Initialise instances of TaskList and Storage

glenda-1506 committed:
Refactor functions in main to use those in Storage and TaskList class

glenda-1506 committed:
Remove redundant variables

glenda-1506 committed:
Import lovespiritualException class in Parser class

glenda-1506 committed:
Add parseCommand function

Handle input parsing through normalising input by trimming and converting it to lowercase

glenda-1506 committed:
Refactor if-else with switch cases and integrate parseCommand function from Parser class

glenda-1506 committed:
Refactor return type from int to void to remove redundant return values

glenda-1506 committed:
Add Storage, TaskList, and UI into main class

glenda-1506 committed:
Remove redundant libraries

glenda-1506 committed:
Add constructor to initialise UI, Storage, and TaskList and load tasks from file

glenda-1506 committed:
Add run() function for main command loop

glenda-1506 committed:
Refactor main function to initialise application with data file and start the command processing loop

glenda-1506 committed:
Add input 'find' to Parser class and integrate 'find' function in switch case

glenda-1506 committed:
Add find function

glenda-1506 committed:
Add Javadoc for main lovespiritual class

glenda-1506 committed:
Add Javadoc for lovespiritual function in lovespiritual class

glenda-1506 committed:
Add Javadoc for run function in lovespiritual class

glenda-1506 committed:
Add Javadoc for main function in lovespiritual class

glenda-1506 committed:
Add Javadoc for event function in lovespiritual class

glenda-1506 committed:
Add Javadoc for deadline function in lovespiritual class

glenda-1506 committed:
Add Javadoc for todo function in lovespiritual class

glenda-1506 committed:
Add Javadoc for unmark function in lovespiritual class

glenda-1506 committed:
Add Javadoc for mark function in lovespiritual class

glenda-1506 committed:
Add Javadoc for Parser class

glenda-1506 committed:
Add Javadoc for Todo class

glenda-1506 committed:
Add Javadoc for Event class

glenda-1506 committed:
Add Javadoc for Deadline class

glenda-1506 committed:
Add Javadoc for Storage class

glenda-1506 committed:
Add Javadoc for Storage class

glenda-1506 committed:
Add Javadoc for lovespiritualException class

glenda-1506 committed:
Add Javadoc for UI class

glenda-1506 committed:
Add Javadoc for Task class

glenda-1506 committed:
Add Javadoc for TaskList class

glenda-1506 committed:
Refactor functions by removing redundant variables

glenda-1506 committed:
Merge pull request #1 from glenda-1506/branch-Level-9

branch-Level-9

glenda-1506 committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/lovespiritual/lovespiritual.java

glenda-1506 committed:
Add Javadoc for find function in lovespiritual class

glenda-1506 committed:
Merge pull request #2 from glenda-1506/branch-A-JavaDoc

branch-A-JavaDoc

glenda-1506 committed:
Tweak list and bye command for more lenient user input

glenda-1506 committed:
Refactor the title of the user guide

glenda-1506 committed:
Add headings for the user guide

glenda-1506 committed:
Add quick start content

glenda-1506 committed:
Comment out default Duke README

glenda-1506 committed:
Refactor quick start format

glenda-1506 committed:
Refactor heading formats

glenda-1506 committed:
Add command format content

glenda-1506 committed:
Add features content

glenda-1506 committed:
Add data saving and editing instructions

glenda-1506 committed:
Add date and time format to deadline task

glenda-1506 committed:
Add date and time format to event task

glenda-1506 committed:
Add error handling for invalid date-time format for deadline and event function

glenda-1506 committed:
Refactor extractTask function to handle date and time formatting

glenda-1506 committed:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java/lovespiritual/task/Deadline.java
#	src/main/java/lovespiritual/task/Event.java

glenda-1506 committed:
Merge pull request #3 from glenda-1506/branch-Level-8

Branch-Level-8

glenda-1506 committed:
Refactor emoticons for cross-platform compatibility

glenda-1506 committed:
Debug output issue in find function

glenda-1506 committed:
Move find function to TaskList class

glenda-1506 committed:
Debug output issue in find function

glenda-1506 committed:
Refactor emoticons for cross-platform compatibility

glenda-1506 committed:
Debug error output in event function

glenda-1506 committed:
Remove redundant class

glenda-1506 committed:
Add error handling for invalid event time

Ensure start date/time is before end date/time

glenda-1506 committed:
Refactor data and time format

glenda-1506 committed:
Edit README for clearer instructions

glenda-1506 committed:
Remove redundant lines in README

glenda-1506 committed:
Refactor error message

glenda-1506 committed:
Add product introduction

glenda-1506 committed:
Add product screenshot

glenda-1506 committed:
Add product screenshot picture and refactor path in README

glenda-1506 committed:
Edit product information
